Sage/carbon
# chore(button): drop classic support FE-2629

### Proposed behaviour
Remove the Classic theme from `Button` to make adding new variants easier.
The Classic button is also used by `Split Button` and `Multi-Action Button`, these components have also had Classic theme removed.



### Current behaviour
`Button`, `Split Button` and `Multi-Action Button` has a classic Theme.

![image](https://user-images.githubusercontent.com/2328042/77561980-5279b580-6eb7-11ea-98db-f7f654bbd799.png)
![image](https://user-images.githubusercontent.com/2328042/77562003-59082d00-6eb7-11ea-868d-a142f6df88d6.png)
![image](https://user-images.githubusercontent.com/2328042/77562025-60c7d180-6eb7-11ea-9cfe-9bb3dff70200.png)
![image](https://user-images.githubusercontent.com/2328042/77562042-69b8a300-6eb7-11ea-9714-542f52f9e6ab.png)
![image](https://user-images.githubusercontent.com/2328042/77562058-6fae8400-6eb7-11ea-90a5-f0ff2888935a.png)
![image](https://user-images.githubusercontent.com/2328042/77562111-81902700-6eb7-11ea-864e-3372e6d8be56.png)
![image](https://user-images.githubusercontent.com/2328042/77562131-86ed7180-6eb7-11ea-9d00-06d0d1c07452.png)
![image](https://user-images.githubusercontent.com/2328042/77562148-8bb22580-6eb7-11ea-95e3-bfb3597e14fa.png)
![image](https://user-images.githubusercontent.com/2328042/77562164-8fde4300-6eb7-11ea-8112-74f86dc4ed76.png)
![image](https://user-images.githubusercontent.com/2328042/77562181-953b8d80-6eb7-11ea-9658-5664fd1d5172.png)
![image](https://user-images.githubusercontent.com/2328042/77562193-9a98d800-6eb7-11ea-9d3c-27dab0606113.png)
![image](https://user-images.githubusercontent.com/2328042/77562231-ab494e00-6eb7-11ea-96e4-b7f5bdb27e38.png)
![image](https://user-images.githubusercontent.com/2328042/77562244-b0a69880-6eb7-11ea-9c98-392fb6a0f167.png)
![image](https://user-images.githubusercontent.com/2328042/77562395-ddf34680-6eb7-11ea-8346-fee28d8bfd7f.png)

![image](https://user-images.githubusercontent.com/2328042/77562314-c3b96880-6eb7-11ea-9048-706463de7041.png)
![image](https://user-images.githubusercontent.com/2328042/77562357-d16eee00-6eb7-11ea-9643-1f2cd4e45749.png)

![image](https://user-images.githubusercontent.com/2328042/77562436-ed728f80-6eb7-11ea-947b-9e41505b7c07.png)
![image](https://user-images.githubusercontent.com/2328042/77562461-f5323400-6eb7-11ea-8af4-e2de25322c66.png)






### Checklist


<del>- [ ] Theme support
<del>- [ ] Typescript `d.ts` file added or updated
### Additional context
I plan on revisiting Button again to make further improvements. This PR is the minimum changes required to remove Classic from Button. Doing so will make it significantly easier to add more variations. 

In a future PR I hope to
- rename the `as` prop to `buttonType`
- simplify button styling
- remove react-router
- prevent `<a type="button" />` when passing a `href`

### Testing instructions

There will be other Classic components that use Button including `Form` and `Dialog` what will now display a Base/Mint/Aegean button instead of Classic. This is to be expected. We no longer maintain the Classic theme for any components and as such our consumers should be migrating to a "modern" theme.

Use `npm start` to start Storybook.

The following stories should **look** the same.
_Please note, all of the knobs are now available, previously some were conditional based on if you had a different knob value. The functionality is the same._

| Local | Production |
|-|-|
| [Button Default][button] | [Button Default][button-prod] |
| [Multi-Action Button Default][multi-action-button] | [Multi-Action Button Default][multi-action-button-prod] |
| [Split Button Default][split-button] | [Split Button Default][split-button-prod] |

The following stories have been removed

| Local | Production |
|-|-|
| [Button Classic][button-classic] | [Button Classic][button-classic-prod] |
| [Button as a Sibling Classic][button-sibling-classic] | [Button as a Sibling Classic][button-sibling-classic-prod] |
| [Split Button Classic][split-button-classic] | [Split Button Classic][split-button-classic-prod] |
| [Multi-Action Button Classic][multi-action-button-classic] | [Multi-Action Button Classic][multi-action-button-classic-prod] |



[button]: http://localhost:9001/?path=/story/button--default
[button-prod]: https://carbon.sage.com/?path=/story/button--default
[button-classic]: http://localhost:9001/?path=/story/button--classic
[button-classic-prod]: https://carbon.sage.com/?path=/story/button--classic
[button-sibling-classic]: http://localhost:9001/?path=/story/button--as-a-sibling-classic
[button-sibling-classic-prod]: https://carbon.sage.com/?path=/story/button--as-a-sibling-classic

[multi-action-button]: http://localhost:9001/?path=/story/multi-action-button--default
[multi-action-button-prod]: https://carbon.sage.com/?path=/story/multi-action-button--default
[multi-action-button-classic]: http://localhost:9001/?path=/story/multi-action-button--classic
[multi-action-button-classic-prod]: https://carbon.sage.com/?path=/story/multi-action-button--classic


[split-button]: http://localhost:9001/?path=/story/split-button--default
[split-button-prod]: https://carbon.sage.com/?path=/story/split-button--default
[split-button-classic]: http://localhost:9001/?path=/story/split-button--classic
[split-button-classic-prod]: https://carbon.sage.com/?path=/story/split-button--classic
